server/src/main/java/org/apache/druid/query/lookup/LookupSerdeModule.java

```diff
@@ -58,6 +59,9 @@ public List<? extends Module> getJacksonModules()
   public void configure(Binder binder)
   {
     JsonConfigProvider.bind(binder, LookupModule.PROPERTY_BASE, LookupConfig.class);
+    //Even though LookupSerdeModule will be used for processes that are not loading lookups, we need to bind LookupNodeService,
+    //so that TaskToolboxFactory and TaskToolbox will get their dependency. Binding it won't load lookups.
```

Comment:
Hmm, I suppose we could have just made this dependency `@Nullable` in `TaskToolboxFactory`.

The only usage is to build a `DruidDiscoveryNode` where this is used to populate the services map.

From the point of service discovery, I think it would be more appropriate to just not add the lookup node service to the map rather than declaring a dummy node service.

Reply:
tried this and got following error for compaction task
````
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Could not find a suitable constructor in org.apache.druid.discovery.LookupNodeService. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at org.apache.druid.discovery.LookupNodeService.class(LookupNodeService.java:38)
  while locating org.apache.druid.discovery.LookupNodeService
    for the 26th parameter of org.apache.druid.indexing.common.TaskToolboxFactory.<init>(TaskToolboxFactory.java:165)
  at org.apache.druid.cli.CliPeon.bindTaskConfigAndClients(CliPeon.java:506) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.apache.druid.cli.CliPeon$1)

1 error
	at org.apache.druid.cli.CliPeon.run(CliPeon.java:421)
	at org.apache.druid.cli.Main.main(Main.java:112)
Caused by: java.lang.RuntimeException: com.google.inject.CreationException: Unable to create injector, see the following errors:
````

Reply:
I'm not too familiar with Guice(using spring boot at work) so might be there is other option to have Optional dependency?

Reply:
actually @Nullable is the way to mark it optional...so I'm not sure why it doesn't work

Reply:
maybe in additional to @nullable need to add empty c'tor to LookupNodeService ?

Reply:
No problem, we can try to address this in a follow up.